src/main/javascript/controller/cat-base-detail-controller.js

```diff
@@ -157,18 +157,29 @@ function CatBaseDetailController($scope, $state, $stateParams, $location, $windo
      * Upon success the view mode of the details of the currently created / updated object will be shown.
      * Upon an error the reported errors (global & field errors) will be shown to the user and the edit mode
      * will remain active.
+     *
+     * * @param {object} stayInEdit If true the view stays in detail edit state after save instead of switching to
+     *                              detail view state.
      */
-    $scope.save = function () {
-        endpoint.save($scope.editDetail).then(function (data) {
+    $scope.save = function (stayInEdit) {
+        // When passing data to an asynchronous method, it makes sense to clone it.
+        endpoint.save(angular.copy($scope.editDetail)).then(function (data) {
             $globalMessages.clearMessages();
             $scope.$fieldErrors = undefined;
-            if (!$scope.exists) {
-                $scope.$broadcast('formReset');
-                $state.go(config.name + '.detail', {id: data.id});
-            } else {
-                $scope.editDetail = undefined;
-                $scope.detail = data;
-                update();
+            if (stayInEdit){
+                $scope.editDetail = data;
+                // Refresh-Breadcrumb:
+                $scope.reloadDetails();
+            }
+            else {
```

Comment:
Done.


Von: Thomas Scheinecker [mailto:notifications@github.com]
Gesendet: Montag, 2. März 2015 09:41
An: Catalysts/cat-angular
Cc: KURZ Michael
Betreff: Re: [cat-angular] -added: optional parameter "stayInEdit" to save() method, so that the co... (#4)


In src/main/javascript/controller/cat-base-detail-controller.js<https://github.com/Catalysts/cat-angular/pull/4#discussion_r25583865>:

>              $globalMessages.clearMessages();

>              $scope.$fieldErrors = undefined;

> -            if (!$scope.exists) {

> -                $scope.$broadcast('formReset');

> -                $state.go(config.name + '.detail', {id: data.id});

> -            } else {

> -                $scope.editDetail = undefined;

> -                $scope.detail = data;

> -                update();

> +            if (stayInEdit){

> +                $scope.editDetail = data;

> +                // Refresh-Breadcrumb:

> +                $scope.reloadDetails();

> +            }

> +            else {

please put the else in the same line as the closing } from the if

—
Reply to this email directly or view it on GitHub<https://github.com/Catalysts/cat-angular/pull/4/files#r25583865>.